graph/src/data/subgraph/mod.rs

```diff
@@ -773,15 +799,14 @@ impl UnresolvedDataSourceTemplate {
 
         info!(logger, "Resolve data source template"; "name" => &name);
 
-        mapping
-            .resolve(resolver, logger)
-            .map(|mapping| DataSourceTemplate {
-                kind,
-                network,
-                name,
-                source,
-                mapping,
-            })
+        let mapping = mapping.resolve(resolver, logger).await?;
+        Ok(DataSourceTemplate {
+            kind,
+            network,
+            name,
+            source,
+            mapping,
+        })
```

Comment:
I like to avoid variable assignments where possible. Would this work?
```rust
Ok(DataSourceTemplate {
    kind,
    network,
    name,
    source,
    mapping: mapping.resolve(resolver, logger).await?,
})
```